compiler/ir/tsl/tiled_stride.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from dataclasses import dataclass
+
+from compiler.ir.tsl.stride import Stride
+
+
+@dataclass
+class TiledStride:
+    """
+    A Tiled Stride is a collection of Strides, one for each level of a Tiled Stride.
+
+    The class works with an arbitrary number of tile levels. The most common use case
+    is one level of tiling, which results in a tile depth of 2.
+
+    For example, consider a Tiled Stride for a vector of length 8 of [1, 16] x [4, 2].
+    This Tiled Stride means that the distance between elements in the first
+    tile of size 4 is 1, and the distance between elements in the second tile
+    of size 2 is 16. This results in the following memory addresses:
+
+    1 - 2 - 3 - 4 - 16 - 17 - 18 - 19
+
+    """
+
+    strides: list[Stride]
+
+    def __init__(self, strides: list[Stride]):
+        self.strides = list(strides)
+
+    def __str__(self) -> str:
+        strides = ", ".join(
+            str(stride.step) if stride.step else "?"
+            for stride in reversed(self.strides)
```

Comment:
Does this mean they are constructed from innermost to outermost? Is this not more confusing?